src/hotspot/share/utilities/ostream.hpp

```diff
@@ -95,10 +100,10 @@ class outputStream : public CHeapObjBase {
    void print_cr(const char* format, ...) ATTRIBUTE_PRINTF(2, 3);
    void vprint(const char *format, va_list argptr) ATTRIBUTE_PRINTF(2, 0);
    void vprint_cr(const char* format, va_list argptr) ATTRIBUTE_PRINTF(2, 0);
-   void print_raw(const char* str)            { write(str, strlen(str)); }
-   void print_raw(const char* str, size_t len)   { write(str,         len); }
-   void print_raw_cr(const char* str)         { write(str, strlen(str)); cr(); }
-   void print_raw_cr(const char* str, size_t len){ write(str,         len); cr(); }
+   void print_raw(const char* str)                { print_raw(str, strlen(str)); }
+   void print_raw(const char* str, size_t len);
+   void print_raw_cr(const char* str)             { print_raw(str); cr(); }
+   void print_raw_cr(const char* str, size_t len) { print_raw(str, len); cr(); }
```

Comment:
Can these be simplified if we have a default value for len? i.e.
```
void print_raw(const char* str, size_t len = strlen(str));
```

Reply:
Interesting question. I try to avoid non-const default arguments since I never can recall their restrictions. 

Looked it up, I think it is not possible to refer to other function arguments in default values:

```
A default argument is evaluated each time the function is called with no argument for the corresponding parameter. Function parameters are not allowed in default arguments except if they are [not evaluated](https://en.cppreference.com/w/cpp/language/expressions#Potentially-evaluated_expressions). Note that parameters that appear earlier in the parameter list are in [scope](https://en.cppreference.com/w/cpp/language/scope):
```

Reply:
Okay - thanks